core/ledger/kvledger/pvtrwstorage/transient_store.go

```diff
@@ -16,26 +16,166 @@ limitations under the License.
+}
+
+// TransientStoreProvider provides an instance of a TransientStore
 type TransientStoreProvider interface {
-	OpenStore(namespace string) (TransientStore, error)
+	OpenStore(ledgerId string) (TransientStore, error)
 	Close()
 }
 
-// EndorserTxRWSets contains the private rwset along with the id of the producing endorser
-type EndorserTxRWSets struct {
-	endorserid string
-	rwset      []byte
+// TransientStore holds an instance of a levelDB.
+type transientStore struct {
+	db       *leveldbhelper.DBHandle
+	ledgerId string
 }
 
-// TransientStore manages the storage of private read-write sets for a namespace for sometime.
-// Ideally, a ledger can remove the data from this storage when it is committed to the permanent storage or
-// the pruning of some data items is enforced by the policy
+type rwsetScanner struct {
+	txId  string
+	dbItr iterator.Iterator
+}
+
+// TransientStore manages the storage of private read-write sets for a ledgerId.
+// Ideally, a ledger can remove the data from this storage when it is committed to
+// the permanent storage or the pruning of some data items is enforced by the policy
 type TransientStore interface {
-	Persist(endorserid string, txRWSet *TxRWSet, endorsementBlkHt uint64) error
-	GetTxRWSetByTxid() []*EndorserTxRWSets
+	// Persist stores the private read-write set of a transaction in the transient store
+	Persist(txid string, endorserid string, endorsementBlkHt uint64, privateSimulationResults []byte) error
+	// GetTxPrivateRWSetbyTxId returns private read-write set of a given transaction
+	GetTxPrivateRWSetByTxId(txId string) (commonledger.ResultsIterator, error)
+	// Purge removes private read-writes set generated by endorsers at block height lesser than
+	// a given maxBlockNumToRetain. In other words, Purge only retains private read-write sets
+	// that were generated at block height of maxBlockNumToRetain or higher.
 	Purge(maxBlockNumToRetain uint64) error
+	// GetMinEndorsementBlkHt returns the lowest retained endorsement block height
 	GetMinEndorsementBlkHt() (uint64, error)
 	Shutdown()
 }
+
+// NewTransientStoreProvider instantiates TransientStoreProvider
+func NewTransientStoreProvider() TransientStoreProvider {
+	dbPath := ledgerconfig.GetTransientStorePath()
+	dbProvider := leveldbhelper.NewProvider(&leveldbhelper.Conf{DBPath: dbPath})
+	return &transientStoreProvider{dbProvider: dbProvider}
+}
+
+// OpenStore returns a handle to a ledgerId in TransientStore
+func (provider *transientStoreProvider) OpenStore(ledgerId string) (TransientStore, error) {
+	dbHandle := provider.dbProvider.GetDBHandle(ledgerId)
+	return &transientStore{db: dbHandle, ledgerId: ledgerId}, nil
+}
+
+// Close closes the TransientStoreProvider
+func (provider *transientStoreProvider) Close() {
+	provider.dbProvider.Close()
+}
+
+// Persist stores the private read-write set of a transaction in the transient store
+func (store *transientStore) Persist(txId string, endorserId string,
```

Comment:
There would be multiple private Write-sets for each tran - one per collection. For instance, if a tran touches 3 different collections. The private write-set for different collections needs to be propagated to the desired participants.
In order to meet the above requirement, please add 'collection' also as one of the parameters in the above API.

We would need to change all the APIs to take a collection